File: app/src/main/java/com/example/amconsofttest/ui/base/BaseFragment.java
package com.example.amconsofttest.ui.base;

import android.widget.Toast;

import com.arellomobile.mvp.MvpAppCompatFragment;
import com.arellomobile.mvp.MvpFragment;
import com.example.amconsofttest.ui.BaseView;

public abstract class BaseFragment extends MvpAppCompatFragment implements BaseView {

    @Override
    public void showProgressBar() {
        DialogUtil.displayProgressDialog(getActivity());
    }

    @Override
    public void hideProgressBar() {
        DialogUtil.hideProgressDialog();
    }

    @Override
    public void showToast(int id) {
        Toast.makeText(getActivity(), getText(id), Toast.LENGTH_LONG).show();
    }

}

File: app/src/main/java/com/example/amconsofttest/ui/base/BaseFragment.kt
package com.example.amconsofttest.ui.base

import android.widget.Toast

import com.arellomobile.mvp.MvpAppCompatFragment
import com.arellomobile.mvp.MvpFragment
import com.example.amconsofttest.ui.BaseView

abstract class BaseFragment : MvpAppCompatFragment(), BaseView {

    override fun showProgressBar() {
        DialogUtil.displayProgressDialog(activity)
    }

    override fun hideProgressBar() {
        DialogUtil.hideProgressDialog()
    }

    override fun showToast(id: Int) {
        Toast.makeText(activity, getText(id), Toast.LENGTH_LONG).show()
    }

}
